core/src/unprocessed_transaction_storage.rs

```diff
@@ -145,17 +146,16 @@ impl UnprocessedTransactionStorage {
         }
     }
 
-    pub fn deserialize_and_insert_batch(
+    pub fn insert_batch(
         &mut self,
-        packet_batch: &PacketBatch,
-        packet_indexes: &[usize],
+        deserialized_packets: Vec<ImmutableDeserializedPacket>,
     ) -> InsertPacketBatchesSummary {
         match self {
             Self::VoteStorage(vote_storage) => {
                 let VoteBatchInsertionMetrics {
                     num_dropped_gossip,
                     num_dropped_tpu,
-                } = vote_storage.deserialize_and_insert_batch(packet_batch, packet_indexes);
+                } = vote_storage.insert_batch(deserialized_packets);
                 InsertPacketBatchesSummary {
```

Comment:
why not have the respective `insert_batch` functions just return the summary struct directly?